acceptancetests charm version: add test for cwd no git and for relative path

nammn opened the pull request:
## Tests still needed to be added
[x] local charm with version file

## Related CHARM PR 
https://github.com/juju/charm/pull/294

## Description of change
add test for cwd no git and for relative path

If the charm folder does not contain git it will do the following:
```
else {
		logger.Debugf("charm is not in revision control directory")
		// Return empty string.
		return "", "", nil <--- which corresponds to the charm version 
	}
```
## QA steps

run the acceptance tests

## Bug reference

https://bugs.launchpad.net/juju/+bug/1848149 

regarding this bug we would need to make further changes as currently a file with a "version" does not get parsed correctly (not at all). The current acceptancetests should add some confidence to the cases except the one I just mentioned. 

Another separate PR should add the fix + another acceptancetest

mitechie commented on tests/suites/cli/use_local_charm.sh:
```diff
@@ -2,23 +2,87 @@
+  fi
+
+
+  if [ "${SHA_OF_NTP}" != "${CURRENT_CHARM_SHA}" ]; then
```
I think this check also needs to be in the other deploy with git.

nammn commented:
```
❯ BOOTSTRAP_REUSE_LOCAL=older ./main.sh cli 

==> Checking for dependencies
==> TEST BEGIN: test_cli
==> Checking for dependencies
====> Reusing bootstrapped juju (2.7)
Added 'test-local-charms' model on localhost/default with credential 'localhost' for user 'admin'
====> Bootstrapped juju (1s)
===> [   ] Running: test deploy local charm revision
====> Reusing bootstrapped juju (2.7)
Added 'local-charm-deploy' model on localhost/default with credential 'localhost' for user 'admin'
====> Bootstrapped juju (0s)
Success: "exit status 128" not found
====> Destroying juju model local-charm-deploy
====> Destroyed juju model local-charm-deploy
===> [ ✔ ] Success: test deploy local charm revision (8s)
===> [   ] Running: test deploy local charm revision no git
====> Reusing bootstrapped juju (2.7)
Added 'local-charm-deploy-no-git' model on localhost/default with credential 'localhost' for user 'admin'
====> Bootstrapped juju (1s)
Success: "charm is not in revision control directory" found
===> [ ✔ ] Success: test deploy local charm revision no git (4s)
===> [   ] Running: test deploy local charm revision relative path
====> Reusing bootstrapped juju (2.7)
Added 'relative-path' model on localhost/default with credential 'localhost' for user 'admin'
====> Bootstrapped juju (1s)
Initialized empty Git repository in /tmp/ci-BJz3KE71tN/.git/
[master (root-commit) 7ee7502] rand_file
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 rand_file
Deploying charm "local:bionic/ntp-0".
====> Destroying juju model relative-path
====> Destroyed juju model relative-path
===> [ ✔ ] Success: test deploy local charm revision relative path (8s)
===> [   ] Running: test deploy local charm revision invalid git
====> Reusing bootstrapped juju (2.7)
Added 'local-charm-deploy-wrong-git' model on localhost/default with credential 'localhost' for user 'admin'
====> Bootstrapped juju (1s)
Initialized empty Git repository in /tmp/ci-dalKoufzQe/.git/
[master (root-commit) 773d1a1] rand_file
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 rand_file
Deploying charm "local:bionic/ntp-0".
====> Destroying juju model local-charm-deploy-wrong-git
====> Destroyed juju model local-charm-deploy-wrong-git
===> [ ✔ ] Success: test deploy local charm revision invalid git (7s)
==> TEST DONE: test_cli (28s)
==> Cleaning up
====> Cleaning up jujus

==> Test result: success
==> Tests Removed: /home/nam/golang/src/github.com/juju/juju/tests/tmp.fHB
==> TEST COMPLETE
```

SimonRichardson commented on tests/suites/cli/use_local_charm.sh:
```diff
@@ -1,49 +1,122 @@
 # Checks whether the cwd is used for the juju local deploy.
-run_deploy_local_charm_revision() {
+test_deploy_local_charm_revision() {
```
Can we keep these as `run_` - these are officially sub tests for `test_local_charms` test, so they shouldn't confuse people

nammn replied:
Oh sry was not on purpose, probably a rebase error by me 👯‍♂

SimonRichardson commented on tests/suites/cli/use_local_charm.sh:
```diff
@@ -1,49 +1,122 @@
 
+# Checks the cwd has no git
+test_deploy_local_charm_revision_no_git() {
+  echo
```
`run_`

SimonRichardson commented on tests/suites/cli/use_local_charm.sh:
```diff
@@ -1,49 +1,122 @@
+  OUTPUT=$(juju deploy --debug . 2>&1)
+
+  check_contains "${OUTPUT}" "charm is not in revision control directory"
+
```
Remove blank line

SimonRichardson commented on tests/suites/cli/use_local_charm.sh:
```diff
@@ -1,49 +1,122 @@
+
+
+# Checks whether the cwd is used for the juju local deploy.
+test_deploy_local_charm_revision_relative_path() {
```
`run_`

SimonRichardson commented on tests/suites/cli/use_local_charm.sh:
```diff
@@ -1,49 +1,122 @@
+
 # CWD with git, deploy charm with git, but -> check that git describe is correct
-run_deploy_local_charm_revision_invalid_git() {
+test_deploy_local_charm_revision_invalid_git() {
```
`run_`

nammn commented:
!!build!!

nammn commented:
$$merge$$
